components/blitz/src/omero/gateway/model/TableData.java

```diff
@@ -0,0 +1,294 @@
+        result = prime * result + (int) numberOfRows;
+        result = prime * result + (int) originalFileId;
+        result = prime * result + Arrays.hashCode(columns);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        TableData other = (TableData) obj;
+        if (originalFileId != other.getOriginalFileId())
+            return false;
+        if (offset != other.getOffset())
+            return false;
+        if (numberOfRows != other.numberOfRows)
+            return false;
+        if (Arrays.hashCode(columns) != Arrays.hashCode(other.columns))
+            return false;
+        if (objectArrayHashCode(data, columns) != objectArrayHashCode(
+                other.data, other.columns))
+            return false;
+        return true;
+    }
+
+    /**
+     * Generates a hash code by iterating over all elements and casting them to
+     * their proposed classes
+     * 
+     * @param objects
+     *            The array to generate the object for
+     * @param types
+     *            The types of the elements in the array; it's assumed that
+     *            every element of a sub array (<code>object[i][]</code>) has
+     *            the same type (<code>types[i]</code>).
+     * @return See above
+     */
+    private int objectArrayHashCode(Object[][] objects,
+            TableDataColumn[] columns) {
+
+        // The reason for this method is, that we can't use Arrays.hashCode()
+        // method on Object[][] arrays, because an Object[][] array can be
+        // for example an array of Object arrays, but also an array of String
+        // arrays, in which case they have a different hash codes and are
+        // *not* equal even if they contain in fact equal elements.
+        // An example for this:
+        // Object[][] test = new String[1][1];
+        // test[0] = new String[1];
+        // test[0][0] = new String("test");
+        // Object[][] test2 = new Object[1][1];
+        // test2[0][0] = new String("test");
+        // --> Arrays.hashCode(test) != Arrays.hashCode(test2)
+
+        final int prime = 31;
+        int result = 1;
+
+        for (int i = 0; i < objects.length; i++) {
+            Object[] col = objects[i];
+
+            for (int j = 0; j < col.length; j++) {
+                Object castedObject = columns[i].getType().cast(col[j]);
+                if (columns[i].getType().isArray())
+                    result = prime * result
+                            + Arrays.hashCode((Object[]) castedObject);
+                else
+                    result = prime * result + castedObject.hashCode();
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```

Comment:
Agreed. Definitely better to move to `StringBuilder` and not let the instance escape a given method.